### PR TITLE
Doc update: migrated to doxygen 1.9.6 and markdown content.

### PR DESCRIPTION
**Pull request by vovamarch**

### CODE REVIEW

File: documentation/doxygen/compiler.dxy.in [hunk 1]
```diff
@@ -187,7 +208,7 @@ SHORT_NAMES            = NO
 # description.)
 # The default value is: NO.
 
-JAVADOC_AUTOBRIEF      = YES
+JAVADOC_AUTOBRIEF      = NO
```
Comment (VladimirUmek) [inline documentation/doxygen/compiler.dxy.in:211]: For the time being I would set this back to YES.

This makes Doxygen to correctly recognize brief descriptions when parsing retarget_fs.h API file for File Interface.
Affects API Reference -> File Interface display.

File: documentation/doxygen/compiler.dxy
Comment (VladimirUmek) [inline documentation/doxygen/compiler.dxy:1]: This file should be deleted.

We don't need this file, gen_doc.sh generates dynamically it from compiler.dxy.in (ideally it should also delete it after the documentation is generated, but I don't think we have this anywhere across repositories).

**Review by VladimirUmek — CHANGES_REQUESTED** [review]

File: documentation/doxygen/compiler.dxy
Comment (vovamarch) [reply documentation/doxygen/compiler.dxy:1]: Removed.

File: documentation/doxygen/compiler.dxy.in (hunk 1 above)
Comment (vovamarch) [reply documentation/doxygen/compiler.dxy.in]: Reverted to YES. But we shall think of changing away from javadoc format.

**Review by VladimirUmek — APPROVED** [review]

File: documentation/doxygen/linkchecker-out.csv
Comment (VladimirUmek) [inline documentation/doxygen/linkchecker-out.csv:1]: This file should probably be removed, right?

**Comment by VladimirUmek** [comment]
Several errors are now reported with the latest changes, I think they should be reviewed.

File: documentation/doxygen/linkchecker-out.csv
Comment (vovamarch) [reply documentation/doxygen/linkchecker-out.csv:1]: Yes, removed now, and is already part of .gitignore.

**Comment by vovamarch** [comment]
"Several errors are now reported with the latest changes, I think they should be reviewed."

There are linkchecker errors for non-public pages that should disappear when they get published. Complains about access timeout to arm.com page is often in GH, needs separate investigation.

**Pull request merged**